src/persist/src/operators/stream.rs

```diff
@@ -827,11 +796,19 @@ mod tests {
             condition_input.send((((), ()), 0, 1));
 
             primary_input.advance_to(1);
+
             condition_input.advance_to(1);
             while seal_probe.less_than(&1) {
                 worker.step();
             }
 
+            // Confirm that both primary and condition listeners receive the seal to 1.
```

Comment:
Alright, I think I did this but I had to weaken one of the invariants (we can't guarantee that the primary seal will always show up after the condition seal because the nondeterminism in select/generally it being two independent channels means that the primary and condition could be swapped). We _can_ assert that the primary is <= the immediately next condition seal if there is one.

I also used the select macro from the crossbeam channel crate ( https://docs.rs/crossbeam/0.8.1/crossbeam/channel/macro.select.html ) instead of the futures select because it wasn't clear to me how to make the futures select fit here.